qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -58,23 +74,44 @@ def __init__(
             delta: the finite difference step size for the gradient computation. It has a minimum
                 value of 1e-5.
             max_iterations: the maximum number of iterations of the AdaptVQE algorithm.
+            gradient: a class that converts operator expression to the first-order gradient based
+                on the method mentioned.
         """
         validate_min("threshold", threshold, 1e-15)
-        validate_min("delta", delta, 1e-5)
+
+        if isinstance(gradient, float):
+            # this scenario can only occur while using the deprecate_arguments wrapper which will
+            # move any argument supplied to delta into gradient.
+            gradient = Gradient(grad_method="fin_diff", epsilon=gradient)
+
+        if gradient is None:
+            #To reproduce the original default case supply delta=1 explicitly.
```

Comment:
I really meant that you should add this information in the documentation string above. That way, it will be visible in the online documentation which will not be the case with a simple comment like this one. You could also be a bit more explicit by saying:
```
To reproduce the default behavior of AdaptVQE prior to Qiskit Nature 0.4 you should supply `delta=1` explicitly.
This will use a finite difference scheme for the gradient evaluation whereas after version 0.4 a parameter shift gradient will be used.
[Maybe insert a link to the Qiskit Terra Gradient documentation here].
```